vm-setup/roles/v1aX_integration_test/tasks/node_reuse.yml

```diff
@@ -175,9 +175,8 @@
     delay: 2
     until:
       - bmhs is succeeded
-      - (saved_bmhs.resources | filter_provisioning("deprovisioning") | map(attribute='metadata.name')) |
-        intersect(bmhs.resources | filter_provisioning("available,ready") | map(attribute='metadata.name')) |
-        length ==  (NUMBER_OF_BMH_MD | int)
+      - saved_bmhs.resources | filter_provisioning("deprovisioning") | json_query('metadata.name') == bmhs.resources | 
+        filter_provisioning("available,ready") | json_query('metadata.name') 
```

Comment:
Can you please update the name of the task too because it is misleading, in fact it is always only one BMH not BMH**s** that is deprovisioning.